### ISSUE TITLE
unable to receive remote notification on devices with iOS11+

### ISSUE THREAD

**Issue by pravnviji** [phonegap/phonegap-plugin-push#2050]
### Expected Behaviour
when the app is running in the foreground, we expect the push notification to be delivered.

### Actual Behaviour
however, the plugin method didreceiveremotenotification is not triggered when the app is in foreground. it works well, if the app is in background or killed.

### Reproduce Scenario (including but not limited to)
1.1. run the app in the foreground
1.2. send push
1.3. push wont be delivered

2.1 put the app in the background
2.2 send push
2.3 push will be delivered 

#### Steps to Reproduce 
we dont see a pattern that can be simulated to reproduce the issue. we do know however, that the issue affects only iOS11+ devices

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
iOS11.0+

#### Cordova CLI version and cordova platform version

$ cordova -version
6.5.0

$ cordova platform version ios
Installed platforms:
  android 5.2.2
  ios 4.2.1

$ cordova plugin version | grep phonegap-plugin-push 
phonegap-plugin-push 1.7.1 "PushPlugin"

#### Sample Push Data Payload
{  
   "title":"Test push notification",
   "message":"Trying to check notification in fg",
   "additionalData":{  
      "full_desc":"Testing app in foreground"
   },
   "aps":{  
      "alert":{  
         "title":"test",
         "body":"iOS push notification"
      }
   }
}

#### Sample Code that illustrates the problem
pushPlugIn: {
        pushObj: undefined,
        pushInit: function () {
            var me = this;
            console.log('push init on added');
            try {
             
                var senderId = myURL.pushNotification.senderId,
                    gcmSandbox = myURL.pushNotification.gcmSandbox;

                var push = PushNotification.init({
                    android: {
                        senderID: senderId
                    },
                    ios: {
                        senderID: senderId,
                        gcmSandbox: gcmSandbox,
                        //true for dev. false for production
                        alert: "true",
                        badge: "true",
                        sound: "true"
                    }
                });
                push.on('notification', function (data) {
                    console.log('check the notification in foreground ------->');
                    console.log('init push ', data);
                });
                push.on('error', function (e) {
                    console.log('Notification error -- > ' + JSON.stringify(e));
                });
                me.pushObj = push;
            }
            catch (e) {
                //push not avl.
                console.log('push not avl.');
            }
        },

var a=MyApp.app.getApplication().getController('AppController.PushNotificaitonController');
a.pushPlugIn.pushInit();

#### Logs taken while reproducing problem

Not able to get log when app is on foreground. push.on('notification', fn(){} function is not getting triggered only on iOS11 and above.

**Comment by macdonst** [comment]
@pravnviji open up the workspace in Xcode and make sure that:

1) Under capabilities, push is enabled
2) The entitlements file matches what you are trying to send to, is it production or development?

**Comment by pravnviji** [comment]
@macdonst 

1) Push is enabled.
2) Trying send notification to production, and certificate is perfect is working for version below 11. **This issue is happen only > iOS11** and rest of version is working as expected

**Comment by macdonst** [comment]
@pravnviji and what does your entitlements file say?

**Comment by pravnviji** [comment]
APS Enivornment - Production.

**Comment by pravnviji** [comment]
Could you confirm because of plugin version ? because i am using 1.7.1.

**Comment by macdonst** [comment]
@pravnviji ah yes, you should upgrade if you want iOS 11 to work correctly.

**Comment by pravnviji** [comment]
@macdonst  Thanks for reply.

I had compared v1.7.1 with v2.1.0, I couldn't find any difference in receiving notification ( foreground ). Is there is any special handling done for iOS 11 and above ?.

**Comment by pawan-mittal** [comment]
@macdonst I am having same issue on iOS 11. When app is on foreground. push.on('notification', fn(){} function is not getting triggered. This is urgent for me too.

**Comment by pawan-mittal** [comment]
@macdonst Thanks for your reply. I got it working for me as local-notification caused the issue. Compatibility issue of local notification and push notification. I used diff plugin for local-notification.
Ref https://github.com/katzer/cordova-plugin-local-notifications.git.

**Comment by macdonst** [comment]
@pawan-mittal yeah, the local notification plugin does not play nice with this one. I should really send a PR to it someday if I get a chance.

**Comment by katzer** [comment]
@macdonst I am the author of the local-notification plugin. What is the reason for the incompatibility?

**Comment by macdonst** [comment]
@katzer it's been awhile since I looked at it but both plugins are trying to listen for notifications and they end up stomping on each other.

**Comment by katzer** [comment]
@macdonst I just took a fast look through the code of phonegap-plugin-push.

Maybe the reason is that this plugin listen for deprecated notification delegates (to support older iOS versions) while the local-notification plugin registers for the new API.

Can imagine that the system wont invoke old and new delegates at same time.

**Comment by harsh-iomedia** [comment]
@katzer 
Can you please come up with a solution so that we can use both the plugins in the same project without any conflicts of notification. Because i want to show the notification stock alert of IOS when the app is in the foreground, but when i install the local-notification it starts conflicting witht he 

Please help us with the possible solution.

**Comment by macdonst** [comment]
PR #2363
